### PR TITLE
swagger updates to address docs and sdk issues 

### PR DESCRIPTION
**Pull request by finiteattractor**
Found few issues while generating docs and SDK. Addressed by these changes. All test pass and SDK generation and test cases passing as well. Changes:
* Fixed wrong enum definitions
* IncludedUpdateClassifications to use comma separated strings
* Fixed parameters not showing in docs
* removed discriminator property, was resulting in missing property in generated models
* First letter capital for better looking doc
- Cleaned common definition file
* fixed x-ms-enum without enum on same level error in code generation
* removed subscriptionid/resourcegroupname/azureautomationname parameters to simplify sdk (they are not in the client)
* Changed scheduleinfo to be a required property
* moved filter parameter for better sdk parameter ordering
* ran auto indent on files

### CODE REVIEW

**Comment by marstr** [comment]
@olydis / @veronicagg it seems like CI didn't run here. Who can I talk to about getting that service to point here?

**Comment by veronicagg** [comment]
@marstr it appears the PR is originating from AzureAutomationTeam which for some reason Travis may permission issues with. I'd recommend starting the PR from the owner's fork. 
@shawns1 any ideas on why this type of thing happens when sent from team organizations?

**Comment by sergey-shandar** [comment]
Similar issue here https://github.com/Azure/azure-rest-api-specs/pull/2112

**Comment by sergey-shandar** [comment]
@finiteattractor could you login to Travis-CI using https://github.com/AzureAutomationTeam account?

File: specification/automation/resource-manager/Microsoft.Automation/2015-10-31/account.json [hunk 1]
```diff
@@ -197,7 +212,12 @@
         },
         "parameters": [
           {
-            "$ref": "./definitions.json#/parameters/ResourceGroupNameParameter"
+            "name": "resourceGroupName",
```
Comment (marstr) [inline specification/automation/resource-manager/Microsoft.Automation/2015-10-31/account.json]: I'm surprised that this change is introducing more code duplication. Any particular reason behind this?

File: specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json [hunk 2]
```diff
@@ -59,38 +59,41 @@
                     "readOnly": true
                 }
             },
-            "required": [ "updateConfiguration" ]
+            "required": [
+                "updateConfiguration",
```
Comment (marstr) [inline specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json]: I believe that @fearthecowboy / @olydis require that we use two-spaced formatting. Please change that back :)

**Review by marstr — CHANGES_REQUESTED** [review]
We'll likely have more feedback once CI has started, but these are a few comments to get things started.

File: specification/automation/resource-manager/Microsoft.Automation/2015-10-31/account.json (hunk 1 above)
Comment (finiteattractor) [reply specification/automation/resource-manager/Microsoft.Automation/2015-10-31/account.json]: Agree, but unfortunately when generating documentation, parameters defined this was were not shown in parameters table in docs. Looking at other working examples that work suggested this change.

File: specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json (hunk 2 above)
Comment (finiteattractor) [reply specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json]: Will do...

File: specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/softwareUpdateConfiguration.json [hunk 3]
```diff
@@ -38,34 +37,41 @@
...
+          "Create software update configuration": {
+            "$ref": "./examples/softwareUpdateConfiguration/createSoftwareUpdateConfiguration.json"
+          }
         },
-        "operationId": "softwareUpdateConfigurations_Create",
+        "operationId": "SoftwareUpdateConfigurations_Create",
         "parameters": [
           {
-            "$ref": "./definitions.json#/parameters/subscriptionId"
+           "$ref": "../2015-10-31/definitions.json#/parameters/SubscriptionIdParameter"
```
[the remark this answers is not in the document]
Comment (finiteattractor) [reply specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/softwareUpdateConfiguration.json]: Yes, the new version only add 3 resources to previous one (and it references other resources from previous version). The linking here is ok as we are just referencing a parameter definition which common accross.

**Comment by marstr** [comment]
Closing in favor of #2139

**Pull request closed without merging**